cortex-r-rt/src/lib.rs

```diff
@@ -214,10 +186,25 @@ macro_rules! save_context {
 /// handler.
 ///
 /// It should match `save_context!`.
-#[cfg(all(
-    any(arm_architecture = "v7-r", arm_architecture = "v8-r"),
-    any(target_abi = "eabihf", feature = "eabi-fpu")
-))]
+#[cfg(not(any(target_abi = "eabihf", feature = "eabi-fpu")))]
+macro_rules! restore_context {
```

Comment:
I think I preferred these as save/restore pairs, because save and restore have to match whilst FPU-save and no-fpu-save don't have to match.

Reply:
yes, they're still pairs right? Or did I mess something up?

Reply:
I think you moved the first restore_context to after the second save_context and it was before?

It was

save A
restore A
save B
restore B

It's now

save A
save B
restore A
restore B

Reply:
ah you just mean the order they appear. I can put this back to the old order, no problem